cache_test.go

```diff
@@ -23,23 +23,27 @@ func TestCaching(t *testing.T) {
 			candidates:          make(chan masquerade, 1000),
 			masquerades:         make(chan masquerade, 1000),
 			maxAllowedCachedAge: 250 * time.Millisecond,
-			maxCacheSize:        2,
+			maxCacheSize:        3,
 			cacheSaveInterval:   50 * time.Millisecond,
 			toCache:             make(chan masquerade, 1000),
+			providers:           testProviders(),
+			defaultProviderID:   testProviderID,
 		}
 		go d.fillCache(make([]masquerade, 0), cacheFile)
 		return d
 	}
 
 	now := time.Now()
-	ma := masquerade{Masquerade{Domain: "a", IpAddress: "1"}, now}
-	mb := masquerade{Masquerade{Domain: "b", IpAddress: "2"}, now}
-	mc := masquerade{Masquerade{Domain: "c", IpAddress: "3"}, now}
+	ma := masquerade{Masquerade{Domain: "a", IpAddress: "1"}, now, testProviderID}
+	mb := masquerade{Masquerade{Domain: "b", IpAddress: "2"}, now, testProviderID}
+	mc := masquerade{Masquerade{Domain: "c", IpAddress: "3"}, now, ""}         // defaulted
+	md := masquerade{Masquerade{Domain: "d", IpAddress: "4"}, now, "sadcloud"} // skipped
 
 	d := makeDirect()
 	d.toCache <- ma
 	d.toCache <- mb
 	d.toCache <- mc
+	d.toCache <- md
 
 	readMasquerades := func() []masquerade {
 		var result []masquerade
```

Comment:
Maybe also add some assertions that after re-reading the masquerades they have the right provider information populated? You might need to use a different defaultProviderID than `testProviderID`.

Reply:
added